src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -82,12 +82,23 @@ internal static TextReader GetOrCreateReader()
             }
             else
             {
-                return StdInReader;
+                return TextReader.Synchronized(StdInReader);
             }
         }
 
-        public static bool KeyAvailable { get { return StdInReader.KeyAvailable; } }
+        [SuppressMessage("Microsoft.Reliability", "CA2002:DoNotLockOnObjectsWithWeakIdentity")]
+        public static bool KeyAvailable
+        {
+            get
+            {
+                lock (Console.In)
+                {
+                    return StdInReader.IsUnprocessedBufferEmpty() || StdInReader.StdinReady;
```

Comment:
The original code here that's being hoisted was:
```C#
return !r.IsUnprocessedBufferEmpty() || r.StdinReady;
```
This code is:
```C#
return StdInReader.IsUnprocessedBufferEmpty() || StdInReader.StdinReady;
```
Is there a reason that `!` was removed from the first condition?  It appears to now be wrong, returning that a key is available if the unprocessed buffer is empty.